docs-src/prod-readiness-context/data-encryption.md

```diff
@@ -0,0 +1,42 @@
+---
+id: data-encryption
+title: How to encrypt data using a custom Data Converter
+description: Using encryption in your custom Data Converter ensures that all your sensitive application data is secure when handled by the Temporal Server.
+sidebar_label: Data Encryption
+tags:
+  - guide-context
+  - production-readiness
+---
+
+Temporal stores and persists the data handled in your Workflow Execution.
+For example, any data that is persisted in the following contexts is stored in the Workflow Execution Event History:
+
+- inputs to your Workflows, Activities, and Child Workflows
+- inputs to your Signals
+- metadata information
+- results of Local Activities, Side Effects
+- Search Attributes
+- Application Errors and Failures.
+
+Encrypting this data at your application-level ensures that all your sensitive application data is secure when handled by the Temporal Server. It also ensures that your data exists unencrypted only on the Client and the Worker process that is executing the Workflows and Activities, on hosts that you control.
+
+To encrypt your data, configure your custom encryption logic with a [`PayloadCodec`](/concepts/what-is-a-payload-codec) and set it with a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
+
+A [`PayloadCodec`](/concepts/what-is-a-payload-codec) transforms your payloads, for example by implementing compression and/or encryption and decryption, and is an optional step that happens between the wire and the [Payload Converter](/concepts/what-is-a-payload-converter):
+
+```bash
+Temporal Server <--> Wire <--> Payload Codec <--> Payload Converter <--> User code
+```
+
+A `PayloadCodec` implementation is applied with a custom Data Converter in your Client options.
+
+You can run your `PayloadCodec` [remotely](/concepts/what-is-remote-data-encoding) with a [Codec Server](/concepts/what-is-a-codec-server), and use the Codec Server endpoints in your WebUI and tctl to decode your encrypted paylods locally.
+
+However, if you plan on setting up remote data encoding for all your payloads, ensure that you consider all security implications of running encryptions remotely before implementing it.
```

Comment:
```suggestion
However, if you plan on setting up Remote Data Encoder, ensure that you consider all security implications of running encryptions remotely before implementing it.
```
Doesn't matter for all or not, the same considerations should be taken just by bringing this infrastructure piece at all.